Zend/zend_builtin_functions.c

```diff
@@ -1857,9 +1858,14 @@ ZEND_API void zend_fetch_debug_backtrace(zval *return_value, int skip_last, int
 				zval *arg = zend_get_zval_ptr(op_data, op_data->op1_type, &op_data->op1, call);
 				if (Z_TYPE_P(arg) == IS_UNDEF) goto not_frameless_call;
 			}
+			zend_function *func = ZEND_FLF_FUNC(opline);
+			/* Assume frameless functions are not recursive with themselves.
+			 * This condition may be true when observers are enabled. */
+			if (last_call && last_call->func == func) {
+				goto not_frameless_call;
+			}
```

Comment:
There will be a frame (the observed call) on top of the FRAMELESS opcode now. If I don't do this, it will try to generate the frame for the FRAMLESS frame on top of that, i.e. observed frameless calls would appear twice in stacktraces now.